Tests/StarknetTests/Providers/ProviderTests.swift

```diff
@@ -396,10 +399,26 @@ final class ProviderTests: XCTestCase {
             classHash: deployAccountTx.classHash
         )
 
-        let simulations2 = try await provider.simulateTransactions([invokeWithoutSignature, deployAccountWithoutSignature], at: .tag(.pending), simulationFlags: [.skipValidate])
+        let simulations2 = try await provider.simulateTransactions([invokeWithoutSignature, deployAccountWithoutSignature], at: .tag(.pending), simulationFlags: [.skipValidate]).send()
 
         XCTAssertEqual(simulations2.count, 2)
         XCTAssertTrue(simulations2[0].transactionTrace is StarknetInvokeTransactionTrace)
         XCTAssertTrue(simulations2[1].transactionTrace is StarknetDeployAccountTransactionTrace)
     }
+
+    func testBatchGetTransactionByHash() async throws {
```

Comment:
1. This test needs to be improved - it should perform calls to get data about different transactions.
2. There should be a test for incorrect responses order. Imho it should be placed in a separate class eg. `JsonRpcResponseTests` as it focuses on the responses parsing mechanism.

Wdyt? @DelevoXDG

Reply:
1. I think that can be enough
2. Yes, should add such test to `JsonRpcResponseTests`

Reply:
I've added [test for incorrect responses order](https://github.com/software-mansion/starknet.swift/blob/85d70fcd517c2766830124e5d6b50ee9d69f7c30/Tests/StarknetTests/Data/JsonRpcResponseTests.swift#L118)